src/Run/SettingValueProvider.cs

```diff
@@ -0,0 +1,134 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.InteropServices;
+using System.IO;
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+
+namespace Microsoft.DotNet.Execute
+{
+    internal static class SettingValueProvider
+    {
+        private static string s_cpuArch = null;
+        private static string s_osName = null;
+        private static string s_osVersion = null;
+        private static Dictionary<string, string> s_osToDefaultVersionMap = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase)
+        {
+            { OS.OSX, "osx.10.10" },
+            { OS.FreeBSD, "osx.10.10" },
+            { OS.NetBSD, "osx.10.10" },
+            { OS.Linux, "ubuntu.14.04" },
+            { OS.Windows, string.Empty }
+        };
+
+        public static string Get(string settingName)
+        {
+            PropertyInfo property = typeof(SettingValueProvider).GetTypeInfo().GetProperty(settingName);
+            return (property != null) ? (string)property.GetValue(null) : string.Empty;
+        }
+        
+        public static string CPUArch
+        {
+            get
+            {
+                if (s_cpuArch == null)
+                {
+                    switch (RuntimeInformation.OSArchitecture)
+                    {
+                        case Architecture.X86:
+                            s_cpuArch = Arch.X86;
+                            break;
+
+                        case Architecture.Arm:
+                            s_cpuArch = Arch.Arm;
+                            break;
+
+                        case Architecture.Arm64:
+                            s_cpuArch = Arch.Arm64;
+                            break;
+
+                        case Architecture.X64:
+                        default:
+                            s_cpuArch = Arch.X64;
+                            break;
+                    }
+                }
+
+                return s_cpuArch;
+            }
+        }
+
+        public static string OSVersion
```

Comment:
I would call this something like OSRid or something so it was clear what it would be used for.

The default value for windows is string.Empty, which feels like a bad default as well.  Should it be "win7" ?